src/panels/config/automation/condition/types/ha-automation-condition-state.ts

```diff
@@ -41,19 +42,36 @@ export class HaStateCondition extends LitElement implements ConditionElement {
         @value-changed=${this._valueChanged}
         allow-custom-value
       ></ha-entity-attribute-picker>
-      <paper-input
-        .label=${this.hass.localize(
-          "ui.panel.config.automation.editor.conditions.type.state.state"
-        )}
-        .name=${"state"}
-        .value=${state}
-        @value-changed=${this._valueChanged}
-      ></paper-input>
+      ${state.map(
+        (value, idx) => html` <paper-input
+          .label=${this.hass.localize(
+            "ui.panel.config.automation.editor.conditions.type.state.state"
+          )}
+          .name=${"state-" + idx}
+          .value=${value}
+          @value-changed=${this._valueChanged}
```

Comment:
Use a different event handler and just only add the index as property. So you don't need code like `if (name.startsWith("state-"))`